### PR TITLE
Handle Back button on Phone mode

### PR DESCRIPTION
**Pull request by felipeerias**
Wolvic's phone UI was not handling the Back button properly, because of some code in VRBrowserActivity which assumed a VR environment. The solution is to register a Back button handler specifically for the phone mode which simply calls Android's usual behaviour.

### CODE REVIEW

File: app/src/hvr/java/com/igalia/wolvic/PlatformActivity.java [hunk 1]
```diff
@@ -46,14 +46,15 @@
 
 import java.util.Calendar;
 
-public class PlatformActivity extends Activity implements SurfaceHolder.Callback {
+public abstract class PlatformActivity extends Activity implements SurfaceHolder.Callback, WidgetManagerDelegate {
```
Comment (svillar) [inline app/src/hvr/java/com/igalia/wolvic/PlatformActivity.java:49]: I know we instantiate `VRBrowserActivity` but this seems unrelated to this fix, maybe we could move it to another PR, or at least a different commit

**Review by svillar — APPROVED** [review]

File: app/src/common/shared/com/igalia/wolvic/VRBrowserActivity.java [hunk 2]
```diff
@@ -288,10 +286,11 @@ protected void onCreate(Bundle savedInstanceState) {
         mBackHandlers = new LinkedList<>();
         mBrightnessQueue = new LinkedList<>();
         mCurrentBrightness = Pair.create(null, 1.0f);
-
         mWidgets = new ConcurrentHashMap<>();
-        mWidgetContainer = new FrameLayout(this);
 
+        super.onCreate(savedInstanceState);
+
+        mWidgetContainer = new FrameLayout(this);
```
[the remark this answers is not in the document]
Comment (felipeerias) [reply app/src/common/shared/com/igalia/wolvic/VRBrowserActivity.java:293]: This change is really about moving `super.onCreate(savedInstanceState);` after the internal fields have been initialized.

`super.onCreate()` can end up calling `pushBackHandler()`, which will fail if `mBackHandlers` has not been initialized.

File: app/src/hvr/java/com/igalia/wolvic/PlatformActivity.java (hunk 1 above)
Comment (felipeerias) [reply app/src/hvr/java/com/igalia/wolvic/PlatformActivity.java:49]: I needed to add `WidgetManagerDelegate` so I could call `pushBackHandler()` and `popBackHandler()`, which are defined in that interface.

And then I had to make the class abstract, so I didn't actually need to implement the interface's methods (which will be implemented by the descendent class `VRBrowserActivity`).

Comment (svillar) [reply app/src/hvr/java/com/igalia/wolvic/PlatformActivity.java:49]: Fair enough.
